### PR TITLE
storage: add bucket listing samples

### PR DESCRIPTION
**Pull request by rakyll**

### CODE REVIEW

File: storage/objects/main.go [hunk 1]
```diff
@@ -96,6 +98,42 @@ func write(client *storage.Client, bucket, object string) error {
...
+		fmt.Println(attrs.Name)
+	}
+	// [END storage_list_files]
+	return nil
+}
+
+func listByPrefix(client *storage.Client, bucket, prefix string) error {
+	ctx := context.Background()
+	// [START storage_list_files_with_prefix]
+	it := client.Bucket(bucket).Objects(ctx, &storage.Query{Prefix: prefix})
```
Comment (rakyll) [inline storage/objects/main.go]: Added a comment explaining prefixes.

File: storage/objects/main.go [hunk 2]
```diff
@@ -96,6 +98,45 @@ func write(client *storage.Client, bucket, object string) error {
...
+	// [END storage_list_files]
+	return nil
+}
+
+func listByPrefix(client *storage.Client, bucket, prefix string) error {
+	ctx := context.Background()
+	// [START storage_list_files_with_prefix]
+	// Prefix is a string that filters results to objects whose
+	// names begin with it. For example, "log-2016" will only list objects
+	// whose names will start with "log-2016".
```
Comment (broady) [inline storage/objects/main.go]: Add another comment that it's useful for emulating directory listings?

This page says
https://cloud.google.com/storage/docs/json_api/v1/objects/list

```
In conjunction with the prefix filter, the use of the delimiter parameter allows the list method to operate like a directory listing, despite the object namespace being flat. For example, if delimiter were set to "/", then listing objects from a bucket that contains the objects "a/b", "a/c", "d", "e", "e/f" would return objects "d" and "e", and prefixes "a/" and "e/".
```

**Comment by rakyll** [comment]
Feel free to merge this once the CI passes.

**Comment by broady** [comment]
@dpebot merge when green, please!

**Pull request merged**